osmomath/decimal.go

```diff
@@ -582,6 +582,12 @@ func BigDecFromDec(d Dec) BigDec {
 	return NewBigDecFromBigIntWithPrec(d.BigInt(), PrecisionDec)
 }
 
+// BigDecFromSDKInt returns the BigDec representation of an sdkInt.
+// Values in any additional decimal places are truncated.
+func BigDecFromSDKInt(i Int) BigDec {
+	return NewBigDecFromBigIntWithPrec(i.BigInt(), 0)
+}
```

Comment:
Note that this is not mutative. See that `BigInt()` reinitializes the underlying buffer.


Let's still keep this, please. However, I'm also wondering if we can make a mutative version that would reuse the input's buffer?

Reply:
Yeah I see, currently it's hard to access `Int.i` directly without `BigInt()`

Reply:
Could you make an issue to consider this in the future and potentially talk to the SDK team about exposing such API?

Reply:
Created an issue for tracking [#17772](https://github.com/cosmos/cosmos-sdk/issues/17772)

Reply:
I tried returning Int.i in my local, performance is improved but the potential risks are unclear